haskell/private/pkg_id.bzl

```diff
@@ -43,7 +43,12 @@ def _new(label, version = None):
     """
     return struct(
         label = label,
-        name = label.name.replace("_", "-"),
+        # the haskell package name is the concatenation of the bazel
+        # package name and the rule name
+
+        # for example:
+        # //things/animal:dog -> things-animal-dog
+        name = (label.package.replace('/', '-') + '-' + label.name).replace("_", "-"),
```

Comment:
@guibou the other methods (`to_string` and `library_name`) already "uniquify" the name by prepending the package and workspace.  GHC allows nonunique package names as long as the package *id* is unique.  So we shouldn't need to change the package name too. 

Furthermore, the package name is user-visible (GHC error messages, haddock index, maybe others) so we should prefer to keep it simple.

We have tests for this in `//tests/package-id-clash`, but only at the level of a haskell_library; I suspect there's somewhere in the code for `haskell_binary`  that's using the name when it should be using the id.